### PR TITLE
Replace applescript with https://github.com/sindresorhus/opn

### PR DESCRIPTION
**Pull request by elliottsj**
This allows opening the Chrome debugger on OS X, Linux, and Windows, and succeeds the previous PR which used [browser-launcher2](https://github.com/benderjs/browser-launcher2) and included a `--dangerouslyDisableChromeDebuggerWebSecurity` option: https://github.com/facebook/react-native/pull/2406

[opn](https://github.com/sindresorhus/opn) is cross-platform and much simpler than browser-launcher2 (since we don't have to manage the opened Chrome instance; the OS will just use the default instance).

### CODE REVIEW

**Comment by elliottsj** [comment]
Woo confirmed working on Windows :raised_hands: . `npm start` (i.e. `packager.sh`) doesn't work but this does:
```shell
node node_modules/react-native/packager/packager.js
```

screenshot 2015-10-13 22 50 25

File: packager/getDevToolsMiddleware.js [hunk 1]
```diff
@@ -11,8 +11,20 @@
 var execFile = require('child_process').execFile;
 var fs = require('fs');
 var path = require('path');
+var opn = require('opn');
```
Comment (ide) [inline packager/getDevToolsMiddleware.js]: alphabetically sort requires

File: packager/packager.js [hunk 2]
```diff
@@ -160,7 +160,13 @@ var server = runServer(options, function() {
   console.log('\nReact packager ready.\n');
 });
 
-webSocketProxy.attachToServer(server, '/debugger-proxy');
+var proxy = webSocketProxy.attachToServer(server, '/debugger-proxy');
+
+function isDebuggerConnected() {
+  // Debugger is connected if the app and at least one browser are connected
+  // to the websocket proxy.
+  return proxy.clientsCount() >= 2;
```
Comment (ide) [inline packager/packager.js]: If you are working on Android and iOS at the same time will this always be true?

Comment (elliottsj) [inline packager/packager.js]: If both devices have debug mode enabled, then no, the debugger won't be connected even though two devices are connected to the WS proxy.

The case I can think of where we might have a problem:
1. Developer opens iOS app in debug mode
2. Chrome automatically opens debugger page
3. Developer opens Android app in debug mode
4. Chrome debugger automatically reloads with Android code
5. Developer closes Chrome debugger
6. Developer reloads iOS app
7. Developer expects Chrome to automatically open debugger, but nothing happens

I'm not sure how common a use case this is, but I think to be safe we may be able to grab the client's user agent when they connect and save it in the `clients` list. Will investigate some more tonight.

File: packager/getDevToolsMiddleware.js (hunk 1 above)
Comment (elliottsj) [inline packager/getDevToolsMiddleware.js]: fixed

**Comment by elliottsj** [comment]
@ide I've just pushed a change to `isDebuggerConnected()`; it now checks for Chrome's user agent string and opens the debugger only if the string is not present in the list of web socket proxy clients.

**Comment by ide** [comment]
Thanks, this is looking pretty good.

File: packager/getDevToolsMiddleware.js [hunk 3]
```diff
@@ -29,20 +41,19 @@ module.exports = function(options) {
...
       console.log('Launching Dev Tools...');
-      execFile(
-        path.join(__dirname, script), [debuggerURL],
-        function(err, stdout, stderr) {
-          if (err) {
-            console.log('Failed to run ' + script, err);
-          }
-          console.log(stdout);
-          console.warn(stderr);
+      opn(debuggerURL, {app: [getChromeAppName()]}, function(err) {
```
Comment (frantic) [inline packager/getDevToolsMiddleware.js]: Looks like in this branch you never `end` the response

Comment (frantic) [inline packager/getDevToolsMiddleware.js]: This causes iOS request to time out and it sends another one, which can open Chrome tab when you are not expecting it.

**Comment by frantic** [comment]
Awesome, this works great! Only one problem with ending the request and I'll pull this in.

**Comment by elliottsj** [comment]
@frantic fixed!

**Comment by frantic** [comment]
@facebook-github-bot import

**Comment by brentvatne** [comment]
:rocket:

**Comment by ide** [comment]
This is awesome, thanks @elliottsj

**Comment by elliottsj** [comment]
Glad to help :smile:

**Pull request closed without merging**